### PR TITLE
Bug Fix - Riot auto joined a public room

### PR DESCRIPTION
**Pull request by giomfo**
#2472

### CODE REVIEW

File: vector/src/main/java/im/vector/fragments/RoomsFragment.java [hunk 1]
```diff
@@ -266,31 +267,30 @@ private void onPublicRoomSelected(final PublicRoom publicRoom) {
...
 
+            // Check whether the room exists to handled the cases where the user is invited or he has joined.
+            // CAUTION: the room may exist whereas the user membership is neither invited nor joined.
             Room room = mSession.getDataHandler().getRoom(publicRoom.roomId, false);
+            if (null != room && room.hasMembership(RoomMember.MEMBERSHIP_INVITE)) {
+                Log.d(LOG_TAG, "onPublicRoomSelected : the user is invited -> display the preview " + getActivity());
+                CommonActivityUtils.previewRoom(getActivity(), roomPreviewData);
+            } else if (null != room && room.hasMembership(RoomMember.MEMBERSHIP_JOIN)) {
+                Log.d(LOG_TAG, "onPublicRoomSelected : the user joined the room -> open the room");
+                Map<String, Object> params = new HashMap<>();
```
Comment (ganfra) [inline vector/src/main/java/im/vector/fragments/RoomsFragment.java]: Use final keyword whenever it suits

File: vector/src/main/java/im/vector/fragments/VectorPublicRoomsListFragment.java [hunk 2]
```diff
@@ -139,31 +140,30 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
                 if (null != publicRoom.roomId) {
                     final RoomPreviewData roomPreviewData = new RoomPreviewData(mSession, publicRoom.roomId, null, publicRoom.getAlias(), null);
 
+                    // Check whether the room exists to handled the cases where the user is invited or he has joined.
+                    // CAUTION: the room may exist whereas the user membership is neither invited nor joined.
                     Room room = mSession.getDataHandler().getRoom(publicRoom.roomId, false);
+                    if (null != room && room.hasMembership(RoomMember.MEMBERSHIP_INVITE)) {
```
Comment (ganfra) [inline vector/src/main/java/im/vector/fragments/VectorPublicRoomsListFragment.java]: This block of code seems very close to the other ones. Refactor if possible ?

File: vector/src/main/java/im/vector/fragments/RoomsFragment.java (hunk 1 above)
Comment (giomfo) [reply vector/src/main/java/im/vector/fragments/RoomsFragment.java]: ok

File: vector/src/main/java/im/vector/fragments/VectorPublicRoomsListFragment.java (hunk 2 above)
Comment (giomfo) [reply vector/src/main/java/im/vector/fragments/VectorPublicRoomsListFragment.java]: I agree with you, but I will not be able to refactor it now :(

**Pull request merged**